pkg/auth/providers/ldap/ldap_provider.go

```diff
@@ -211,10 +222,16 @@ func (p *ldapProvider) getLDAPConfig() (*v3.LdapConfig, *x509.CertPool, error) {
 	storedLdapConfig := &v3.LdapConfig{}
 
 	if p.samlSearchProvider() && ldapConfigKey[p.providerName] != "" {
-		if subLdapConfig, ok := storedLdapConfigMap[ldapConfigKey[p.providerName]]; ok {
-			storedLdapConfigMap = subLdapConfig.(map[string]interface{})
+		subLdapConfig, ok := storedLdapConfigMap[ldapConfigKey[p.providerName]]
+		if !ok {
+			return nil, nil, errNotConfigured
 		}
+
+		storedLdapConfigMap = subLdapConfig.(map[string]interface{})
 		mapstructure.Decode(storedLdapConfigMap, storedLdapConfig)
+		if len(storedLdapConfig.Servers) != 1 {
```

Comment:
should we only return not configured error if `if len(storedLdapConfig.Servers) < 1 ` and let ldap.Connect return error if `len(storedLdapConfig.Servers) > 1`?

Reply:
not my logic, grabbed from this: https://github.com/rancher/rancher/blob/master/pkg/auth/providers/common/ldap/ldap_util.go#L42-L44

Reply:
yeah but since we're returning "NotConfigured" error, here we could return it if no server input was provided whereas NewLDAPConn stays as is, but not an important change :)